components/file-browser/FileBrowser.tsx

```diff
@@ -219,101 +282,41 @@ export default function FileBrowser({
-                                variant="outline"
-                                hasArrow
-                              >
-                                <Text _hover={{ cursor: "help" }}>
-                                  {new Intl.DateTimeFormat("en-US", {
-                                    dateStyle: "medium",
-                                    timeStyle: "short",
-                                  }).format(new Date(item.last_modified))}
-                                </Text>
-                              </Tooltip>
-                            ) : (
-                              <Text>&mdash;</Text>
-                            )}
-                          </Td>
-                        )}
-                        {fileBrowser.view.columns.includes("size") && (
-                          <Td>
-                            {item.size ? (
-                              <Tooltip
-                                label={`${item.size} B`}
-                                variant="outline"
-                                hasArrow
-                              >
-                                <Text _hover={{ cursor: "help" }}>
-                                  {item.size && readable(item.size)}
-                                </Text>
-                              </Tooltip>
-                            ) : (
-                              <Text>&mdash;</Text>
-                            )}
-                          </Td>
-                        )}
-                        {isSource && (
-                          <Td>
-                            {endpoint &&
-                              endpoint.https_server &&
-                              item.type === "file" && (
-                                <IconButton
-                                  as="a"
-                                  aria-label="Open"
-                                  href={`${endpoint.https_server}${lsResponse.absolute_path}${item.name}`}
-                                  target="_blank"
-                                  rel="noopener noreferrer"
-                                  size="xs"
-                                  icon={<Icon as={ArrowUpOnSquareIcon} />}
-                                />
-                              )}
-                          </Td>
-                        )}
                       </Tr>
+                    )}
+                    {items.map((item, i) => (
+                      <FileEntry
+                        key={i}
+                        item={item}
+                        isSource={isSource}
+                        httpsServer={endpoint?.https_server}
+                        absolutePath={lsResponse.absolute_path}
+                        includeSize={fileBrowser.view.columns.includes("size")}
+                        includeLastModified={fileBrowser.view.columns.includes(
+                          "last_modified",
+                        )}
```

Comment:
Use the File Browser context in the `<FileEntry>` component instead of passing properties of the context as props.